### PR TITLE
CIRCSTORE-158 add BE validation for update patron notice policies

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose
Resolves: CIRCSTORE-158

1. Create BE validation for update patron notice policies
2. Create tests to test validation

## Approach
1. Update `PatronNoticePoliciesAPI` with new validation for `putPatronNoticePolicyStoragePatronNoticePoliciesByPatronNoticePolicyId`
2. Update `PatronNoticePoliciesApiTest` to test validation changes

## Links
https://issues.folio.org/browse/CIRCSTORE-158

### CODE REVIEW

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 1]
```diff
@@ -296,6 +296,144 @@ public void cannotDeleteInUsePatronNoticePolicy() throws InterruptedException, M
...
+      .put("templateId", UUID.randomUUID().toString())
+      .put("format", "Email")
+      .put("realTime", "true")
+      .put("sendOptions", sendOptions);
+
+    JsonObject noticePolicy = createNoticePolicyWithLoanNotice(loanNotice);
+    JsonResponse response = putPatronNoticePolicy(noticePolicy);
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson().getJsonArray("errors").getJsonObject(0).getString("message"),
+      is("sendHow should not be present"));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Please change error messages to be more informative

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 2]
```diff
@@ -296,6 +296,144 @@ public void cannotDeleteInUsePatronNoticePolicy() throws InterruptedException, M
...
+
+    JsonObject sendOptions = new JsonObject()
+      .put("sendWhen", "Renewed")
+      .put("sendBy", sendBy);
+
+    JsonObject noticePolicy = createNoticePolicyWithSendOptions(sendOptions);
+    JsonResponse response = putPatronNoticePolicy(noticePolicy);
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson().getJsonArray("errors").getJsonObject(0).getString("message"),
+      is("sendBy should not be present"));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Please change error message to be more informative

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 3]
```diff
@@ -296,6 +296,144 @@ public void cannotDeleteInUsePatronNoticePolicy() throws InterruptedException, M
...
+
+    JsonObject sendOptions = new JsonObject()
+      .put("sendWhen", "Renewed")
+      .put("sendEvery", sendEvery);
+
+    JsonObject noticePolicy = createNoticePolicyWithSendOptions(sendOptions);
+    JsonResponse response = putPatronNoticePolicy(noticePolicy);
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson().getJsonArray("errors").getJsonObject(0).getString("message"),
+      is("sendEvery should not be present"));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Please change error message to be more informative

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 4]
```diff
@@ -296,6 +296,144 @@ public void cannotDeleteInUsePatronNoticePolicy() throws InterruptedException, M
...
+      .put("format", "Email")
+      .put("realTime", "true")
+      .put("frequency", "Recurring")
+      .put("sendOptions", sendOptions);
+
+    JsonObject noticePolicy = createNoticePolicyWithLoanNotice(loanNotice);
+    JsonResponse response = putPatronNoticePolicy(noticePolicy);
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson().getJsonArray("errors").getJsonObject(0).getString("message"),
+      is("frequency should not be present"));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Please change error message to be more informative

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java [hunk 5]
```diff
@@ -296,6 +296,144 @@ public void cannotDeleteInUsePatronNoticePolicy() throws InterruptedException, M
...
+      .put("format", "Email")
+      .put("realTime", "true")
+      .put("frequency", "One time")
+      .put("sendOptions", sendOptions);
+
+    JsonObject noticePolicy = createNoticePolicyWithLoanNotice(loanNotice);
+    JsonResponse response = putPatronNoticePolicy(noticePolicy);
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getJson().getJsonArray("errors").getJsonObject(0).getString("message"),
+      is("sendEvery should not be present"));
```
Comment (kostyantyn-kh) [inline src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Please change error message to be more informative

File: src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java (hunk 1 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/api/PatronNoticePoliciesApiTest.java]: Thank you for review, updated

**Pull request merged**